pkgs/development/tools/continuous-integration/gitlab-runner/default.nix

```diff
@@ -32,6 +32,8 @@ buildGoPackage rec {
     sha256 = "0n8hcj2b1pb95x4bd7fb9ri43vgc4h2dj2v3iiziw2imqjyphfx4";
   };
 
+  patches = [ ./v1-fix-shell-path.patch ];
```

Comment:
`exec`'s [`LookPath`](https://golang.org/pkg/os/exec/#LookPath) seems to do what we need here. in case it errors (e.g. `b.Shell` not found in `$PATH`), it falls back to the original logic.
far from the perfect solution, but a quick fix for the patch could look something like this: 
```patch
diff --git a/shells/bash.go b/shells/bash.go
index 839b7781..2b478e1e 100644
--- a/shells/bash.go
+++ b/shells/bash.go
@@ -7,6 +7,7 @@ import (
 	"gitlab.com/gitlab-org/gitlab-ci-multi-runner/common"
 	"gitlab.com/gitlab-org/gitlab-ci-multi-runner/helpers"
 	"io"
+	"os/exec"
 	"path"
 	"runtime"
 	"strconv"
@@ -208,7 +209,11 @@ func (b *BashShell) GetConfiguration(info common.ShellScriptInfo) (script *commo
 	if info.User != "" {
 		script.Command = "su"
 		if runtime.GOOS == "linux" {
-			script.Arguments = append(script.Arguments, "-s", "/bin/"+b.Shell)
+			shellPath, err := exec.LookPath(b.Shell)
+			if err != nil {
+				shellPath = "/bin/"+b.Shell
+			}
+			script.Arguments = append(script.Arguments, "-s", shellPath)
 		}
 		script.Arguments = append(script.Arguments, info.User)
 		script.Arguments = append(script.Arguments, "-c", shellCommand)
```